python-sdk/nuscenes/eval/lidarseg/render.py

```diff
@@ -0,0 +1,327 @@
+    pcl_path = os.path.join(nusc.dataroot, pointsensor['filename'])
+
+    # Load the ground truth labels for the point cloud.
+    gt_path = os.path.join(nusc.dataroot, nusc.get('lidarseg', sd_token)['filename'])
+    gt = LidarSegPointCloud(pcl_path, gt_path)
+    gt.labels = mapper.convert_label(gt.labels)  # Map the labels as necessary.
+
+    # Load the predictions for the point cloud.
+    preds_path = os.path.join(lidarseg_preds_folder, sd_token + '_lidarseg.bin')
+    preds = LidarSegPointCloud(pcl_path, preds_path)
+
+    # Do not compare points which are ignored.
+    ignored_points_idxs = np.where(gt.labels != mapper.ignore_class['index'])[0]
+    gt.labels = gt.labels[ignored_points_idxs]
+    gt.points = gt.points[ignored_points_idxs]
+    preds.labels = preds.labels[ignored_points_idxs]
+    preds.points = preds.points[ignored_points_idxs]
+
+    # Init axes.
+    fig, axes = plt.subplots(1, 3, figsize=(10 * 3, 10), sharex='all', sharey='all')
+
+    # Render ground truth and predictions.
+    gt.render(mapper.coarse_colormap, mapper.coarse_name_2_coarse_idx_mapping, ax=axes[0])
+    preds.render(mapper.coarse_colormap, mapper.coarse_name_2_coarse_idx_mapping, ax=axes[1])
+
+    # Render errors.
+    id2color_for_diff_bev = {0: (191, 41, 0, 255),  # red: wrong label
+                             1: (50, 168, 82, 255)}  # green: correct label
+    colors_for_diff_bev = colormap_to_colors(id2color_for_diff_bev, {0: 0, 1: 1})
+    mask = np.array(gt.labels == preds.labels).astype(int)  # Convert array from bool to int.
+    axes[2].scatter(gt.points[:, 0], gt.points[:, 1], c=colors_for_diff_bev[mask], s=dot_size)
+    axes[2].set_title('Errors (Correct: Green, Mislabeled: Red)')
+
+    # Limit visible range for all subplots.
+    plt.xlim(-axes_limit, axes_limit)
+    plt.ylim(-axes_limit, axes_limit)
+
+    if out_path:
+        plt.savefig(out_path)
+
+    plt.show()
+
+
+if __name__ == '__main__':
+    # Settings.
+    parser = argparse.ArgumentParser(description='Evaluate nuScenes-lidarseg results.')
+    parser.add_argument('--result_path', type=str,
+                        help='The path to the results folder.')
+    parser.add_argument('--out_path', type=str,
+                        help='Folder to store result metrics, graphs and example visualizations.')
+    parser.add_argument('--eval_set', type=str, default='val',
+                        help='Which dataset split to evaluate on, train, val or test.')
+    parser.add_argument('--dataroot', type=str, default='/data/sets/nuscenes',
+                        help='Default nuScenes data directory.')
+    parser.add_argument('--version', type=str, default='v1.0-trainval',
+                        help='Which version of the nuScenes dataset to evaluate on, e.g. v1.0-trainval.')
+    parser.add_argument('--verbose', type=bool, default=False,
+                        help='Whether to print to stdout.')
+    args = parser.parse_args()
+
+    result_path_ = args.result_path
```

Comment:
I think it should be fine passing args to the functions below?

Reply:
Oh, I usually do it this way in case the names of the args change; then I don't need to change every single instance where I use the affected names in `main`, haha